subsys/net/lib/wifi_mgmt_ext/wifi_mgmt_ext.c

```diff
@@ -24,6 +24,60 @@ LOG_MODULE_REGISTER(wifi_mgmt_ext, CONFIG_WIFI_MGMT_EXT_LOG_LEVEL);
 		     "CONFIG_WIFI_CREDENTIALS_STATIC_SSID required");
 #endif /* defined(CONFIG_WIFI_CREDENTIALS_STATIC) */
 
+static int __stored_creds_to_params(struct wifi_credentials_personal *creds,
+				struct wifi_connect_req_params *params)
+{
+	char *ssid = NULL;
+	char *psk = NULL;
+
+	/* SSID */
+	ssid = (char *)k_malloc(creds->header.ssid_len + 1);
+	if (!ssid) {
+		LOG_ERR("Failed to allocate memory for SSID\n");
+		return -ENOMEM;
+	}
+	memset(ssid, 0, creds->header.ssid_len + 1);
+	strncpy(ssid, creds->header.ssid, creds->header.ssid_len);
+	params->ssid = ssid;
+	params->ssid_length = creds->header.ssid_len;
+
+	/* PSK (optional) */
+	if (creds->password_len > 0) {
+		psk = (char *)k_malloc(creds->password_len + 1);
+		if (!psk) {
+			LOG_ERR("Failed to allocate memory for PSK\n");
+			return -ENOMEM;
+		}
+		memset(psk, 0, creds->password_len + 1);
+		strncpy(psk, creds->password, creds->password_len);
+		params->psk = psk;
+		params->psk_length = creds->password_len;
+	}
+
+	/* Defaults */
+	params->security = WIFI_SECURITY_TYPE_PSK;
+	params->mfp = WIFI_MFP_OPTIONAL;
+	params->channel = WIFI_CHANNEL_ANY;
+	params->timeout = CONNECTION_TIMEOUT;
+
+	/* Security type (optional) */
+	if (creds->header.type <= WIFI_SECURITY_TYPE_MAX) {
+		params->security = creds->header.type;
+	} else {
```

Comment:
The current default is overwritten in both branches, so, if you make this as default then `else` branch can be removed.

Reply:
Done